force_bdss/mco/optimizer_engines/base_optimizer_engine.py

```diff
@@ -76,17 +88,21 @@ def _score(self, input_point):
         This is also useful for the testing purposes, when the `evaluate`
         method is mocked.
         """
-        score = self.single_point_evaluator.evaluate(input_point)
+
+        # Calculate and cache the raw KPI values
+        kpi_values = self.single_point_evaluator.evaluate(input_point)
+        self._kpi_cache[tuple(input_point)] = kpi_values
+
+        # Return the score to be minimized
+        score = self._minimization_score(kpi_values)
         log.info("Objective score: {}".format(score))
         return score
 
```

Comment:
This is the heart of it, and as it is never overridden, we don't need to change the code anywhere else (?).

Reply:
Yup, anything that sits on top of the `BaseOptimizerEngine` can always access the raw KPI values from the cache. However, the `_score` function returns the actual objective function (with the optimization objectives taken care of). Therefore the external wrappers using the `IOptmizer` interface only need to care about minimizing some function.